Add AI engine for spam (#11319)
decidim/decidim: decidim-ai/README.md

# Decidim::Ai

The Decidim::Ai is a library that aims to provide Artificial Intelligence tools for Decidim. This plugin has been initially developed aiming to analyze the content and provide spam classification using Naive Bayes algorithm.
All AI related functionality provided by Decidim should be included in this same module.

For more documentation on the AI tools API, please refer to [documentation](https://docs.decidim.org/en/develop/develop/ai_tools.html)

## Installation

In order to install use this module, you need at least Decidim 0.30 to be installed.

To install this module, run in your console:

```bash
bundle add decidim-ai
```

After that, add an initializer file as presented in the [documentation](https://docs.decidim.org/en/develop/services/aitools.html#_configuration)

> After the configuration is added, you need to run the below command, so that the reporting user is created.

Reviewer: For not repeating "After" two times 
```suggestion
Then, you need to run the below command, so that the reporting user is created.
```

# Decidim::Ai

The Decidim::Ai is a library that aims to provide Artificial Intelligence tools for Decidim. This plugin has been initially developed aiming to analyze the content and provide spam classification using Naive Bayes algorithm.
All AI related functionality provided by Decidim should be included in this same module.

For more documentation on the AI tools API, please refer to [documentation](https://docs.decidim.org/en/develop/develop/ai_tools.html)

## Installation

In order to install use this module, you need at least Decidim 0.30 to be installed.

> Add this line to your application's Gemfile:

Author: fixed in 631625c756

# Decidim::Ai

The Decidim::Ai is a library that aims to provide Artificial Intelligence tools for Decidim. This plugin has been initially developed aiming to analyze the content and provide spam classification using Naive Bayes algorithm.

> All AI related functionality provided by Decidim should be included in this same module.

Reviewer: Add link to the docs final URL (we can do this after this is merged, but adding the comment so I don't forget about it)

Author: Fixed in 85481f86e9df65124297e5a61dcba2296576264c

## Installation

In order to install use this library, you need at least Decidim 0.25 to be installed.

Add this line to your application's Gemfile:

```ruby
gem "decidim-ai"
```

And then execute:

```bash
bundle install
```

After that, add an initializer file inside your project, having the following content:

```ruby

> # config/initializers/decidim_ai.rb

Reviewer: Can you check my comment @alecslupu ?

> # config/initializers/decidim_ai.rb

Author: fixed in 85481f86e9df65124297e5a61dcba2296576264c

```

After the configuration is added, you need to run the below command, so that the reporting user is created.

```ruby
bundle exec rake decidim:ai:create_reporting_user
```

> If you have an existing installation, you can use the below command to train the engine with your existing data:

Author: Yes.

> If you have an existing installation, you can use the below command to train the engine with your existing data:

Reviewer: Ok, then can you update the message there? 

```suggestion
Then you can use the below command to train the engine with the module dataset:
```

```ruby

> bundle exec rake decidim:spam:train:moderation

Reviewer: I think at least for the first few iterations of this module, it could be better to have it as separate tasks to make it easier to debug. We can change it afterwards (specially once we have a good `load_module_dataset`). 

Marking it as solved as I understand it'd be a nice to have.
